File: 1256-rank-transform-of-an-array/rank-transform-of-an-array.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        int[] demo = arr.clone();
        Arrays.sort(demo);
        Map<Integer,Integer> lol = new HashMap<>();
        int r=1;
        for(int i:demo){
            if(!lol.containsKey(i)){
                lol.put(i,r);
                r++;
            }
        }
        int[] res = new int[arr.length];
        for(int i =0;i<arr.length;i++){
            res[i] = lol.get(arr[i]);
        }
    return res;
    }
}

File: 1256-rank-transform-of-an-array/rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        se = sorted(list(set(arr)))
        rank_map = {}
        i = 1
        for num in se:
            rank_map[num] = i
            i += 1
        result = []
        for num in arr:
            result.append(rank_map[num])
        return result